orbstation/spells/eldritch_transform.dm

```diff
@@ -0,0 +1,50 @@
+/// Turn into an eldritch creature, many of these have their own spells so this is pretty good value as utility
+/datum/action/cooldown/spell/shapeshift/eldritch_monster
+	name = "Eldritch Becoming"
+	desc = "Endure a horrific transformation into a creature of darkness for a time to use their unnatural abilities. \
+		Once you've made your choice, it cannot be changed."
+
+	background_icon_state = "bg_ecult"
+	button_icon_state = "soultap"
+
+	sound = 'sound/magic/demon_consume.ogg'
+	cooldown_time = 30 SECONDS
+	cooldown_reduction_per_rank = 3 SECONDS
+
+	invocation = "Someone does a weird gesture." // Overriden in before cast
+	invocation_self_message = span_notice("Your flesh unravels and reknits itself!")
+	invocation_type = INVOCATION_EMOTE
+	spell_requirements = SPELL_REQUIRES_NO_ANTIMAGIC
+
+	possible_shapes = list(
+		/mob/living/simple_animal/shade, // You can move really fast and ventcrawl I guess, beware of the chaplain's soul shard!
+		/mob/living/simple_animal/hostile/construct/wraith/mystic,
+		/mob/living/simple_animal/hostile/construct/artificer/mystic,
+		/mob/living/simple_animal/hostile/construct/juggernaut/mystic,
+		/mob/living/simple_animal/hostile/heretic_summon/raw_prophet,
+		/mob/living/simple_animal/hostile/heretic_summon/rust_spirit,
+		/mob/living/simple_animal/hostile/heretic_summon/ash_spirit
+	)
```

Comment:
If possible, these should get tooltips. I don't know if the parent code allows for that though.

Reply:
Would require an upstream refactor, is a good candidate for an upstream refactor. Will add to my list.